commands/install.js

```diff
@@ -0,0 +1,95 @@
+'use strict'
+
+const { spawn } = require('child_process')
+const { getValue } = require('../lib/config')
+const { queryReadline } = require('../lib/util')
+const details = require('./details')
+const {
+  COLORS,
+  line,
+  failure
+} = require('../lib/ncm-style')
+const { helpHeader } = require('../lib/help')
+const chalk = require('chalk')
+const L = console.log
+
+module.exports = install
+module.exports.optionsList = optionsList
+
+async function install (argv, arg1, arg2, arg3) {
+  const childArgv = Array.from(argv)
+  let name
+  let version
+  if (!arg1) {
+    printHelp()
+    process.exitCode = 1
+    return
+  } else if (arg1.indexOf('@') > 0 && !arg2 && !arg3) {
+    childArgv.splice(childArgv.indexOf(arg1), 1)
+    ;[name, version] = arg1.split('@')
+  } else if (arg2 === '@' && arg3) {
+    name = arg1
+    version = arg3
+    childArgv.splice(childArgv.indexOf(arg1), 1)
+    childArgv.splice(childArgv.indexOf(arg2), 1)
+    childArgv.splice(childArgv.indexOf(arg3), 1)
+  } else if (arg1) {
+    name = arg1
+    version = arg2 || 'latest'
+    childArgv.splice(childArgv.indexOf(arg1), 1)
+    if (arg2) childArgv.splice(childArgv.indexOf(arg2), 1)
+  } else {
+    printHelp()
+    process.exitCode = 1
+    return
+  }
+
+  await details(argv, arg1, arg2, arg3)
+
+  const good = process.exitCode === 0 || process.exitCode === undefined
+
+  const confirm = good ? `${COLORS.green} (Y/n)` : `${COLORS.red} (y/N)`
+
+  L(line('|➔', chalk`Install this module? {${confirm}}`, COLORS.yellow))
+  L()
+
+  const choice = (await queryReadline(chalk`{${COLORS.light1} > }`)).trim().toLowerCase()
+
+  if ((good && choice === '') || choice === 'y') {
+    const args = [getValue('installCmd'), `${name}@${version}`, ...childArgv]
```

Comment:
should `installCmd` and `installBin` default to `npm install` ?

Reply:
They already do - in `lib/config.js`. I don't think we should try to fix broken user config for this, though.